app/src/ereferrals/java/org/eyeseetea/malariacare/strategies/DashboardActivityStrategy.java

```diff
@@ -39,6 +72,83 @@ public void onCreate() {
+                mConnectivity, mediaRepository,
+                new Callback() {
+                    @Override
+                    public void onCancel(Exception ex) {
+
+                        //Need to complete credentials (ack from user first time)
+                        if (ex instanceof UserRecoverableAuthIOException) {
+                            DashboardActivity.dashboardActivity.startActivityForResult(
+                                    ((UserRecoverableAuthIOException) ex).getIntent(),
+                                    REQUEST_AUTHORIZATION);
+                            return;
+                        }
+
+                        //Real connection google error
+                        if (ex instanceof GooglePlayServicesAvailabilityIOException) {
+                            showDialog(
+                                    ((GooglePlayServicesAvailabilityIOException) ex)
+                                            .getConnectionStatusCode());
+                            return;
+                        }
+                    }
+
+                    @Override
+                    public void showToast(String message) {
+                        Toast.makeText(mDashboardActivity.getApplicationContext(), message,
+                                Toast.LENGTH_LONG).show();
+                    }
+
+                    @Override
+                    public void acquireGooglePlayServices() {
+                        /**
+                         * Attempt to resolve a missing, out-of-date, invalid or disabled Google
+                         * Play Services installation via a user dialog, if possible.
+                         */
+                        GoogleApiAvailability apiAvailability =
+                                GoogleApiAvailability.getInstance();
+                        final int connectionStatusCode =
+                                apiAvailability.isGooglePlayServicesAvailable(
+                                        PreferencesState.getInstance().getContext());
+                        if (apiAvailability.isUserResolvableError(connectionStatusCode)) {
+                            showDialog(connectionStatusCode);
+                        }
+                    }
+
+                    @Override
+                    public void onSuccess(HashMap<String, String> syncedFiles) {
+                        int correctSyncedFiles = mediaRepository.updateSyncedFiles(syncedFiles);
+                        if (correctSyncedFiles > 0) {
+                            showToast(String.format("%d files synced", correctSyncedFiles));
+                        }
+                    }
+
+                    @Override
+                    public void onRemove(String uid) {
+                        System.out.println(
+                                "downloadFile error (file extension not supported)" + uid);
+                        mediaRepository.deleteModel(mediaRepository.findByUid(uid));
```

Comment:
a class of presentation layer should not invoke mediaRepository, this is a use case responsibility